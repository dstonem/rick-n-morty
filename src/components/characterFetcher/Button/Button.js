export default function Button({ handleClick }) {
  return (
    <button data-testid="button" onClick={handleClick}>
      Fetch Character
    </button>
  );
}
