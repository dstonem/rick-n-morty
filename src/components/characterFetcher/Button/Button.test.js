import React from "react";
import { cleanup, render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";

afterEach(cleanup);

const props = {
  handleClick: jest.fn(),
};

describe("Button", () => {
  it("renders correctly", () => {
    const { getByText } = render(<Button {...props} />);
    expect(getByText("Fetch Character")).toBeTruthy();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Button {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("fires the props.handleClick event onClick", () => {
    render(<Button {...props} />);
    const btn = screen.getByTestId("button");
    fireEvent.click(btn);
    expect(props.handleClick).toHaveBeenCalledTimes(1);
  });
});
