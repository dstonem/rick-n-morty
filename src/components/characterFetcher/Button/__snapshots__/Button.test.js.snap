// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    data-testid="button"
  >
    Fetch Character
  </button>
</DocumentFragment>
`;
